src/core/Entry.cpp

```diff
@@ -325,25 +325,28 @@ QString Entry::totp() const{
 
         int rc = oath_init();
 
-        rc = oath_base32_decode(secret, seedBytes.size(),
-                                &secret_decoded, &secret_decoded_len);
+        if (rc == OATH_OK) {
+            rc = oath_base32_decode(secret, seedBytes.size(),
+                                    &secret_decoded, &secret_decoded_len);
+        }
 
         if (rc == OATH_OK) {
             rc = oath_totp_generate(secret_decoded, secret_decoded_len, t_now, 
                 OATH_TOTP_DEFAULT_TIME_STEP_SIZE, // 30
                 OATH_TOTP_DEFAULT_START_TIME, // 0
                 numDigits, output);
         } else {
+            oath_done();
```

Comment:
oath_done is never called twice, however its called even if oath_init fails.
This method should be refactored.